app/models/story.rb

```diff
@@ -61,4 +61,21 @@ def as_json(options = {})
   def readonly?
     !accepted_at_changed? && accepted_at.present?
   end
+
+  # Set the project start date to today if the project start date is nil
+  # and the state is changing to any state other than 'unstarted' or 'unscheduled'
+  def fix_project_start_date
+    return unless self.state_changed?
+    if self.project && !self.project.start_date && !['unstarted', 'unscheduled'].include?(self.state)
+      self.project.start_date = Date.current
+    end
+  end
+
+  # If a story's 'accepted at' date is prior to the project start date,
+  # the project start date should be moved back accordingly
+  def fix_story_accepted_at
+    if self.accepted_at_changed? && self.accepted_at && self.accepted_at < self.project.start_date
+      self.project.start_date = self.accepted_at
+    end
+  end
 end
```

Comment:
It probably makes sense, since "model" was being referenced all over the method. That said, I'm trying to understand the intent of `LegacyFixes`. It looks like code the project wants to get rid of some time in the future, so wouldn't it make sense to keep it in the same place and make it easier to move/delete/whatever?